src/message/mimebody.rs

```diff
@@ -110,6 +114,66 @@ impl SinglePart {
         SinglePartBuilder::new()
     }
 
+    /// Directly create a `SinglePart` from a file content
+    pub fn attachment_content<T: IntoBody>(body: T, filename: &str, content_type: Mime) -> Self {
+        Self::builder()
+            .header(ContentType(content_type))
+            .header(ContentDisposition {
+                disposition: DispositionType::Attachment,
+                parameters: vec![DispositionParam::Filename(
+                    Charset::Ext("utf-8".into()),
+                    None,
+                    filename.as_bytes().into(),
+                )],
+            })
+            .body(body)
+    }
+
+    /// Directly create a `SinglePart` from a file content
+    pub fn attachment(file: &str, content_type: Mime) -> Result<Self, io::Error> {
+        let path = Path::new(file);
+        let image = fs::read(path)?;
+        let image_body = Body::new(image);
+        Ok(Self::attachment_content(
+            image_body,
+            path.file_name()
```

Comment:
This would force the attachment name to be the same as the local file name, which seems to me like an unnecessary restriction.